openslides_backend/action/group/delete.py

```diff
@@ -0,0 +1,36 @@
+from typing import Any, Dict
+
+from ...models.models import Group
+from ...shared.exceptions import ActionException
+from ...shared.patterns import FullQualifiedId
+from ..default_schema import DefaultSchema
+from ..generics import DeleteAction
+from ..register import register_action
+
+
+@register_action("group.delete")
+class GroupDeleteAction(DeleteAction):
+    """
+    Action to delete a group.
+    """
+
+    model = Group()
+    schema = DefaultSchema(Group()).get_delete_schema()
+
+    def update_instance(self, instance: Dict[str, Any]) -> Dict[str, Any]:
+        """
+        Check for default group of meeting or superadmin group of a meeting.
+        """
+        group = self.datastore.get(
+            FullQualifiedId(self.model.collection, instance["id"]),
+            ["default_group_for_meeting_id", "superadmin_group_for_meeting_id"],
+        )
+        if group.get("default_group_for_meeting_id"):
+            raise ActionException(
+                f"Group {instance['id']} is a default group for a meeting."
+            )
+        if group.get("superadmin_group_for_meeting_id"):
+            raise ActionException(
+                f"Group {instance['id']} is a superadmin group for a meeting."
+            )
+        return instance
```

Comment:
This can also be done via the `models.yml`, just set these fields to `on_delete: PROTECT`. Remove this and then test again that it works as intended

Reply:
Only issue with this is that the error messages become too broad, this was already addressed in https://github.com/OpenSlides/openslides-backend-coordination/issues/96#issuecomment-720973153 and will be fixed with a general solution

Reply:
Okay, we want it more implicitly by the model definition. Okay, done this, Check was okay.
Add PR https://github.com/OpenSlides/OpenSlides/pull/5697 for the model changes.